src/check-update.ts

```diff
@@ -0,0 +1,250 @@
+  DENO_SLACK_HOOKS, // should be the only one needed once the get-scripts hook is supported
+  DENO_SLACK_BUILDER, // but just in case, we can look for builder and runtime as well!
+  DENO_SLACK_RUNTIME,
+];
+
+interface UpdateResponse {
+  update: boolean;
+  breaking: boolean;
+  message: string;
+  error: {
+    message: string;
+  };
+}
+
+interface VersionMap {
+  [key: string]: {
+    current: string;
+    latest: string;
+    update: boolean;
+    breaking: boolean;
+    error: string;
+  };
+}
+
+export const checkForSDKUpdates = async () => {
+  const versionMap: VersionMap = await createVersionMap();
+  const updateResp = createUpdateResp(versionMap);
+  return updateResp;
+};
+
+/**
+ * createVersionMap creates an object that contains each dependency,
+ * featuring information about the current and latest versions, as well
+ * as if breaking changes are present and if any errors occurred during
+ * version retrieval.
+ */
+async function createVersionMap() {
+  const versionMap: VersionMap = await readProjectDependencies();
+
+  // Check each dependency for updates, classify update as breaking or not,
+  // craft message with information retrieved, and note any error that occurred.
+  for (const [sdk, value] of Object.entries(versionMap)) {
+    if (value) {
+      const current = versionMap[sdk].current;
+      let latest = "", error = null;
+
+      try {
+        latest = await fetchLatestModuleVersion(sdk);
+      } catch (err) {
+        error = err;
+      }
+
+      const update = current !== latest;
+      const breaking = hasBreakingChange(current, latest);
+
+      versionMap[sdk] = {
+        ...versionMap[sdk],
+        latest,
+        update,
+        breaking,
+        error,
+      };
+    }
+  }
+
+  return versionMap;
+}
+
+/** readProjectDependencies reads from possible dependency files
+ * (import_map.json, slack.json) and maps them to the versionMap
+ * containing each dependency's update information
+ */
+async function readProjectDependencies(): Promise<VersionMap> {
+  const cwd = Deno.cwd();
+  const versionMap: VersionMap = {};
+
+  // Find SDK component versions in import map, if available
+  const map = await getJson(`${cwd}/import_map.json`);
+  for (const sdkUrl of Object.values(map.imports) as string[]) {
+    for (const sdk of IMPORT_MAP_SDKS) {
+      if (sdkUrl.includes(sdk)) {
+        versionMap[sdk] = {
+          latest: "",
+          update: false,
+          breaking: false,
+          error: "",
+          current: extractVersion(sdkUrl),
+        };
+      }
+    }
+  }
+
+  // Find SDK component versions in slack.json, if available
+  const { hooks }: { [key: string]: string } = await getJson(
+    `${cwd}/slack.json`,
+  );
+  for (const command of Object.values(hooks)) {
+    // TODO :: Does not cover the case where multiple commands use the same SDK.
+    // Only the "last" key containing the relevant SDK will be checked as-is
+    for (const sdk of SLACK_JSON_SDKS) {
```

Comment:
Per your "can we clean this code up?" comment above, this `SLACK_JSON_SDKS` variable should collapse down to a single entry, so I think we can get rid of this outer loop.